### DISCUSSION TITLE
Add extra parameter for useQuery and useMutation

### DISCUSSION THREAD

**Discussion by EVNIKHIL** [TanStack/query#5182] [General]
I have an app with global loader, showing spinner while fetching with help of useIsFetching hook with a predicate function. I have a form with email field with async validation for checking unique email. And there is a local loader. My issue I don't want the global loader here, but because I am checking for some query is fetching in global loader component it will trigger and global loader will show in addition to field loader. I was trying to loop the query cache and conditionally checking by queryKey of all current query by useQueryClient hook but I am not able to loop this properly. Is there any mistake in my way of doing? Is there any better solution for this?

**Comment by TkDodo** [comment]
you can use the `meta` field to tag which queries you'd want to include or exclude, as that is also available in the predicate function.

**Reply by EVNIKHIL** [reply]
Thank you for the reply. I will try this and let you know if worked.
